numpy/lib/npyio.py

```diff
@@ -1145,21 +1148,37 @@ def savetxt(fname, X, fmt='%.18e', delimiter=' ', newline='\n', header='',
         if len(header) > 0:
             header = header.replace('\n', '\n' + comments)
             fh.write(asbytes(comments + header + newline))
-        if iscomplex_X:
-            for row in X:
-                row2 = []
-                for number in row:
-                    row2.append(number.real)
-                    row2.append(number.imag)
-                fh.write(asbytes(format % tuple(row2) + newline))
-        else:
-            for row in X:
-                try:
-                    fh.write(asbytes(format % tuple(row) + newline))
-                except TypeError:
-                    raise TypeError("Mismatch between array dtype ('%s') and "
-                                    "format specifier ('%s')"
-                                    % (str(X.dtype), format))
+
+        # If rows are short we want to format/write up to maxel elemnts at once.
```

Comment:
On Wednesday 30 September 2015, Joseph Martinot-Lagarde wrote:
> Typo, `elemnts` is missing a `e`

Since adding this `e` overflows 80 chars I've added some more 
comments :)

 # If rows are short we want to format/write up to maxel elements at
 # once (but at least one row, e.g ncol elements).